akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -165,6 +165,22 @@ private[akka] class TLSActor(settings: ActorMaterializerSettings,
     params.setServerNames(Collections.singletonList(serverName))
   }
 
+  private def cloneSSLParameters(parameters: SSLParameters): SSLParameters = {
+    val clone = new SSLParameters()
+
+    clone.setCipherSuites(parameters.getCipherSuites)
+    clone.setProtocols(parameters.getProtocols)
+    clone.setWantClientAuth(parameters.getWantClientAuth)
+    clone.setNeedClientAuth(parameters.getNeedClientAuth)
+    clone.setEndpointIdentificationAlgorithm(parameters.getEndpointIdentificationAlgorithm)
+    clone.setAlgorithmConstraints(parameters.getAlgorithmConstraints)
+    clone.setServerNames(parameters.getServerNames)
+    clone.setSNIMatchers(parameters.getSNIMatchers)
+    clone.setUseCipherSuitesOrder(parameters.getUseCipherSuitesOrder)
+
+    clone
+  }
```

Comment:
I'll be touching this area in a PR in the next hour or so, will apply this hint – thanks @2beaucoup!